From 14496e28b54b07a9e565c6a2aeb92244111c60fd Mon Sep 17 00:00:00 2001
From: Pawan Gupta <pawan.kumarx.gupta@intel.com>
Date: Thu, 9 Jun 2016 16:22:44 -0700
Subject: [PATCH 420/639] net: Merge Puma6 and Puma7

Merged puma6 specific changes with puma7.
Puma6 changes are under CONFIG_ARCH_GEN3.

Change-Id: Ib57c18c541713823b1110b1fea7ba5873112868a
Signed-off-by: Pawan Gupta <pawan.kumarx.gupta@intel.com>
---
 net/bridge/br_forward.c           | 2 ++
 net/ipv4/ipconfig.c               | 3 +++
 net/netfilter/nf_conntrack_core.c | 6 +++++-
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/net/bridge/br_forward.c b/net/bridge/br_forward.c
index 6062e4c..0798291 100644
--- a/net/bridge/br_forward.c
+++ b/net/bridge/br_forward.c
@@ -19,7 +19,9 @@
 #include <linux/skbuff.h>
 #include <linux/if_vlan.h>
 #include <linux/netfilter_bridge.h>
+#ifdef CONFIG_APPCPU_GW_PP_HANDLE
 #include <linux/ti_hil.h>
+#endif
 #include "br_private.h"
 
 static int deliver_clone(const struct net_bridge_port *prev,
diff --git a/net/ipv4/ipconfig.c b/net/ipv4/ipconfig.c
index efa1138..c8a2be8 100644
--- a/net/ipv4/ipconfig.c
+++ b/net/ipv4/ipconfig.c
@@ -198,6 +198,9 @@ static bool __init ic_is_init_dev(struct net_device *dev)
 	return user_dev_name[0] ? !strcmp(dev->name, user_dev_name) :
 	    (!(dev->flags & IFF_LOOPBACK) &&
 	     (dev->flags & (IFF_POINTOPOINT|IFF_BROADCAST)) &&
+#if defined(CONFIG_X86_PUMA6) && defined(CONFIG_UDMA)
+	     strncmp(dev->name, "eth_udma", 8) &&
+#endif
 	     strncmp(dev->name, "dummy", 5));
 }
 
diff --git a/net/netfilter/nf_conntrack_core.c b/net/netfilter/nf_conntrack_core.c
index 2dcb382..906cd6f 100644
--- a/net/netfilter/nf_conntrack_core.c
+++ b/net/netfilter/nf_conntrack_core.c
@@ -382,7 +382,11 @@ EXPORT_SYMBOL_GPL(nf_ct_delete);
 
 static void death_by_timeout(unsigned long ul_conntrack)
 {
+#if defined(CONFIG_APPCPU_GW_PP_HANDLE) && defined(CONFIG_MRPC_CONNTRACK_CLIENT)
         nf_ct_delete((struct nf_conn *)ul_conntrack, 0, DEATH_BY_TIMEOUT);
+else
+	nf_ct_delete((struct nf_conn *)ul_conntrack, 0, 0);
+#endif
 }
 
 static inline bool
@@ -1358,13 +1362,13 @@ found:
 	return ct;
 }
 
-
 void nf_ct_iterate_cleanup(struct net *net,
 			   int (*iter)(struct nf_conn *i, void *data),
 			   void *data, u32 portid, int report)
 {
 	struct nf_conn *ct;
 	unsigned int bucket = 0;
+
 	while ((ct = get_next_corpse(net, iter, data, &bucket)) != NULL) {
 		/* Time to push up daises... */
 		if (del_timer(&ct->timeout))
-- 
2.10.1

